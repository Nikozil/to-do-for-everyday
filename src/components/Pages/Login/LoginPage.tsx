import React from 'react';
import { useHistory } from 'react-router';
import { useAuth } from '../../../api/useAuth';
import useInput from '../../../hooks/useInput';
import { useRequireAuth } from '../../../hooks/useRequireAuth';

const Login = () => {
  const auth = useAuth();
  const email = useInput('');
  const password = useInput('');
  const history = useHistory();
  const submitForm = async (event: React.FormEvent<HTMLFormElement>) => {
    await auth.signin(email, password)(event);
    history.push('/');
  };
  return (
    <div>
      <form onSubmit={submitForm}>
        <input placeholder="Email" {...email} />
        <input placeholder="Password" type="password" {...password} />
        <button type="submit">Sign up</button>
      </form>
    </div>
  );
};

export default Login;
