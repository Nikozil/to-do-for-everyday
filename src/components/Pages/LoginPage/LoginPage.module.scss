.login-page {
  display: flex;
  height: 100vh;
  align-items: center;
  justify-content: center;
  background-color: #354052;
}
.login-page__content {
  min-width: 200px;
  max-width: 400px;
  width: 100%;
  margin: 0 5px 0 5px;
  color: #fff;
}
.login-page__logo {
  display: flex;
  justify-content: center;
}
