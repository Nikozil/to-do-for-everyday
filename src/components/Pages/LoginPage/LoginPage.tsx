import React from 'react';
import LoginForm from '../../Forms/LoginForm/LoginForm';
import styles from './LoginPage.module.css';
import Logo from '../../../assets/Logo.png';
import LogoComponent from '../../../assets/LogoComponent/LogoComponent';

const LoginPage: React.FC = () => {
  return (
    <div className={styles.wrapper}>
      <div className={styles.content}>
        <LogoComponent className={styles.logo} />
        <LoginForm />
      </div>
    </div>
  );
};

export default LoginPage;
