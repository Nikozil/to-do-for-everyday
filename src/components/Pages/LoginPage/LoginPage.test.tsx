import { render } from '../../../utils/tests/test-utils';
import LoginPage from './LoginPage';

describe('Login Page tests', () => {
  it('render correct', () => {
    const { asFragment } = render(<LoginPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
