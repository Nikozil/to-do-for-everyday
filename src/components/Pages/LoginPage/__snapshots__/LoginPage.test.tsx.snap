// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page tests render correct 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="content"
    >
      <div
        class="logo"
      >
        <img
          alt="Logo"
          src="Logo.png"
        />
      </div>
      <form
        action="#"
      >
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="email"
          >
            Email
          </label>
          <input
            class="form-control"
            id="email"
            name="email"
            placeholder="guest@gmail.com"
            type="email"
            value=""
          />
          <div
            class="form-text"
            style="height: 21px;"
          />
        </div>
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="password"
          >
            Password
          </label>
          <input
            class="form-control"
            id="password"
            name="password"
            placeholder=""
            type="password"
            value=""
          />
          <div
            class="form-text"
            style="height: 21px;"
          />
        </div>
        <div
          class="mb-1 form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              name="remember"
              type="checkbox"
              value="false"
            />
            Запомнить сессию
          </label>
        </div>
        <div
          class="text-center mb-1"
        >
          <button
            class="btn btn-success center-block"
            type="submit"
          >
            Войти
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
